src/Build/Evaluation/Expander.cs

```diff
@@ -4242,6 +4250,44 @@ private static bool TryGetArg(object[] args, out string arg0)
                 return arg0 != null;
             }
 
+            private static readonly Dictionary<string, StringComparison> StringComparisonNames =
```

Comment:
A drop in an ocean but still, it's wasteful to create the dictionary on each class init when it's used only for evaluating one specific function. Can you please initialize it lazily? Also, shouldn't it be named something like ```s_stringComparisonNames``` to match the convention in other private static data in this class?

Reply:
It happens once per process, probably not noticeable. But, I replaced the dictionary with another approach that was already being used in the same set of overloads.